components/Collection.vue

```diff
@@ -338,6 +338,21 @@ export default {
 		initiallyCollapsed: {
 			type: Boolean,
 			default: false
+		},
+		mapOptions: {
+			default: function() { return {} },  // It's not possible to specify defaults for the individual properties, therefore this is handled in a computed property, and that function is easier when this prop is never `undefined`, hence the empty object (in a non-arrow factory function!).
```

Comment:
Is this code doing something or could we remove it and just keep the comment?

Reply:
Yes it does something: If the prop is not given at all, i.e. undefined, this line set's it to `{}`. This makes the `leafletOptions` function simpler because I can then do this:

https://github.com/Open-EO/openeo-vue-components/blob/9c1f53883c5f870a44410957d95d18beb2ecb058/components/Collection.vue#L384

without having to care about the possibility that `this.mapOptions` may be undefined.

I tried to express exactly that with: "that function is easier when this prop is never `undefined`, hence the empty object"

Reply:
Oh, interesting. Better add a clear comment (could simply be 'Don't remove!' or so) that this shouldn't be removed and has a functionality otherwise I'm pretty sure this would get removed as it looks like it does nothing ;-)

Reply:
Well I thought I wrote something just like that, but looks like I failed to bring the point across :sweat_smile:

Reply:
Yeah, the last part of the comment I just couldn't understand ;-) Is better now!